nb2/service/quote_service.py

```diff
@@ -0,0 +1,58 @@
+from nb2 import db
+from nb2.models import Person, Quote
+from nb2.service.dtos import AddQuoteDTO
+from nb2.service.exceptions import EmptyRequiredFieldException, PersonDoesNotExistException, QuoteAlreadyExistsException
+
+
+def add_quote_to_person(data: AddQuoteDTO):
+    """
+    Add a Quote to a Person's quotes.
+
+    IMPORTANT: This function will not automatically create Person if they
+    don't already exist. It is the caller's responsibility to ensure that
+    a Person with slack_user_id exists before trying to add a Quote to
+    them using this function.
+
+    Required Args:
+        data: An AddQuoteDTO instance.
+
+    Returns:
+        Newly created Quote on success.
+
+    Raises:
+        PersonDoesNotExistException if no Person with slack_user_id exists.
+        QuoteAlreadyExistsException if trying to add a Quote containing content
+        that already exists in one of Person's Quotes.
+
+    Notes:
+        - Timestamp (created) is populated automatically by the db to default
+          to datetime.utc now
+    """
+    empty_fields = data.get_empty_fields()
+    if empty_fields:
+        raise EmptyRequiredFieldException(
+            f"Can't add Quote to Person with these fields empty: {empty_fields}"
+        )
+
+    target_person = Person.query.filter_by(slack_user_id=data.slack_user_id).one_or_none()
+
+    if not target_person:
+        raise PersonDoesNotExistException
+    if target_person.has_said(data.content):
+        raise QuoteAlreadyExistsException
+
+    new_quote = Quote()
+    new_quote.content = data.content
+    new_quote.person_id = target_person.id
+
+    db.session.add(new_quote)
+    db.session.commit()
+    db.session.refresh(new_quote)
+
+    return new_quote
+
```

Comment:
🦆

Reply:
Dunno how I missed that many blank lines haha. Guess I could've used the linters